rules/zend2-rocket/src/Rector/ClassMethod/RenderZendViewRector.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Zend2Rocket\Rector\ClassMethod;
+
+use PhpParser\Node;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+
+ * @see \Rector\Zend2Rocket\Tests\Rector\ClassMethod\RenderZendViewRector\RenderZendViewRectorTest
+ */
+final class RenderZendViewRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('call 'currentZendViewResult' in all Action-methods that do not call 'setNoRender'', [
+            new CodeSample(
+                <<<'PHP'
+class SomeController {
+    public function getArticleInfoAction()
+    {
+    }
+}
+PHP
+,
+                <<<'PHP'
+class SomeController {
+    public function getArticleInfoAction()
+    {
+        return $this->currentZendViewResult();
+    }
+}
+PHP
+
+            )
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [\PhpParser\Node\Stmt\ClassMethod::class];
+    }
+
+    /**
+     * @param \PhpParser\Node\Stmt\ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        // identify if classmethod is an Actionmethod
+        if (! $this->controllerMethodAnalyzer->isAction($node)) {
+            return $node;
+        }
+
+        // check if classmethod calls setNoRender
+        $subnodes = $node->getSubNodeNames();
+        if ( in_array('setNoRender', $subnodes)) {
+            return $node;
+        }
```

Comment:
@TomasVotruba
this was my first approach of checking if $this->setNoRender(); gets called somewhere within the action-method, but this does not work. Do you know any simple approach like this instead of diving into Walking the AST with PHP-Parser

Reply:
`getSubNodeNames()` is rather internal method that gives you list of properties in the node (in the object, not real properties), e.g. https://github.com/nikic/PHP-Parser/blob/a167aa206198570fc8957f7b3f2c6778b0a0e2ad/lib/PhpParser/Node/Expr/Cast.php#L10-L24

<br>

To find method call inside class method, you can do something like this:

```php
$hasSetNodeRenderMethodCall =$this->betterNodeFinder->findFirst((array) $classMethod->stmts, function (Node $node): bool {
    if (! $node instanceof MethodCall) {
        return false;
	} 

    return $this->isName($node->name, 'setNoRender');
});
```

Reply:
great thanks. didn't heard about the `betterNodeFinder` before.